tests/ga/test_extension.py

```diff
@@ -1961,5 +2054,241 @@ def test_ext_handler_with_systemd_cgroup_enabled(self, *args):
+        value_to_delete = "test_value"
+        name_to_keep = "keep_variable"
+        value_to_keep = "keep_value"
+
+        handler_i.set_handler_env_variable({name_to_delete: value_to_delete})
+        handler_i.set_handler_env_variable({name_to_keep: value_to_keep})
+
+        with patch.object(CGroupConfigurator.get_instance(), "start_extension_command", return_value='ok') \
+                as patch_start_cmd:
+
+            handler_i.remove_key_from_handler_env(name_to_delete)
+            handler_i.enable()
+            _, kwargs = patch_start_cmd.call_args
+
+            self.assertTrue(name_to_delete not in kwargs['env'], "The variable should be removed from handler_env list")
+            self.assertIn(name_to_keep, kwargs['env'], "Variable missing from handler_env")
+            self.assertEqual(value_to_keep, kwargs['env'][name_to_keep], "Env Variable Value mismatch")
+
+        # Retrying the removal operation to ensure that no exception is thrown if key not found
+        try:
+            handler_i.remove_key_from_handler_env(name_to_delete)
+        except:
+            self.fail("Removing a key that's not present shouldn't throw an error")
+
+    @patch('time.sleep', side_effect=lambda _: mock_sleep(0.001))
+    def test_failed_env_variables_should_be_set_from_within_extension_commands(self, *args):
+        """
+        This test will test from the perspective of the extensions command weather the env variables are
+        being set for those processes
+        """
+
+        test_file_name = "testfile.sh"
+        update_file_name = test_file_name + " -update"
+        install_file_name = test_file_name + " -install"
+        old_handler_i = TestExtensionUpdateOnFailure._get_ext_handler_instance('foo', '1.0.0')
+        new_handler_i = TestExtensionUpdateOnFailure._get_ext_handler_instance(
+            'foo', '1.0.1',
+            handler={"updateCommand": update_file_name, "installCommand": install_file_name},
+            continue_on_update_failure=True
+        )
+
+        # Script prints env variables passed to this process and prints all starting with AZURE_
+        test_file = """
+            printenv | grep AZURE_
+            """
+
+        self.create_script(file_name=test_file_name, contents=test_file,
+                           file_path=os.path.join(new_handler_i.get_base_dir(), test_file_name))
+
+        with patch.object(new_handler_i.logger, 'verbose', autospec=True) as mock_verbose:
+            # Since we're not mocking the azurelinuxagent.common.cgroupconfigurator..handle_process_completion,
+            # both disable.cmd and uninstall.cmd would raise ExtensionError exceptions and set the DISABLE_FAILED and
+            # UNINSTALL_FAILED env variables.
+            # For update and install we're running the script above to print all the env variables starting with AZURE_
+            # and verify accordingly if the corresponding env variables are set properly or not
```

Comment:
let's add a small comment explaining why the variables show in the verbose log

Reply:
Done